tests/utils/browser.js

```diff
@@ -0,0 +1,27 @@
+const fs = require('fs');
+const feedBackDocs = async (testName = 'allLogs') => {
+  const feedBackDocs = await browser.executeAsync(feedBackDocsScript);
+  if (feedBackDocs && feedBackDocs.flat().length > 0) {
+    fs.writeFileSync(`./tests/logs/feedbackDocs-${testName}.txt`, JSON.stringify(feedBackDocs, null, 2));
+    return true;
+  }
+};
+
+
+const feedBackDocsScript = async (done) => {
+  //This is running inside the browser. indexedDB and PouchDB is available there.
+  // eslint-disable-next-line no-undef
+  const allDbList = await indexedDB.databases();
+  const metaDbList = allDbList.filter(db => db.name.includes('pouch_medic-user') && db.name.endsWith('-meta'));
+  done(Promise.all(metaDbList.map(async (db) => {
+    const nameStripped = db.name.replace('_pouch_', '');
+    // eslint-disable-next-line no-undef
+    const metaDb = new PouchDB(nameStripped);
+    const docs = await metaDb.allDocs({ include_docs: true, startkey: 'feedback-', endkey: 'feedback-\ufff0' });
+    return docs.rows;
+  })));
```

Comment:
This section is a little confusing I think because it's mixing async/await and Promises. As far as I can tell you shouldn't need the `Promise.all` at all because you `await` the result of the only async call. You'd only need Promise.all when you are returning a Promise from the map function, here you're only returning an array (`docs.rows`).

I think this will just work:

```suggestion
  done(metaDbList.map(async (db) => {
    const nameStripped = db.name.replace('_pouch_', '');
    // eslint-disable-next-line no-undef
    const metaDb = new PouchDB(nameStripped);
    const docs = await metaDb.allDocs({ include_docs: true, startkey: 'feedback-', endkey: 'feedback-\ufff0' });
    return docs.rows;
  }));
```

Reply:
I thought in this case the async keyword is wrapping the response from `.map` into a promise. When I just pass that directly to the done callback it doesn't seem to pass that promise to my caller. The return of `executeAsync` becomes an array with an empty object. 



Also tried, 
```
  //This is running inside the browser. indexedDB and PouchDB is available there.
  // eslint-disable-next-line no-undef
  const allDbList = await indexedDB.databases();
  const metaDbList = allDbList.filter(db => db.name.includes('pouch_medic-user') && db.name.endsWith('-meta'));
  return metaDbList.map(async (db) => {
    const nameStripped = db.name.replace('_pouch_', '');
    // eslint-disable-next-line no-undef
    const metaDb = new PouchDB(nameStripped);
    const docs = await metaDb.allDocs({ include_docs: true, startkey: 'feedback-', endkey: 'feedback-\ufff0' });
    return docs.rows;
  });
  ```
  With the caller being 
  
  `const feedBackDocs = await browser.execute(feedBackDocsScript);` and same thing.

Reply:
That's really weird because [the documentation](https://v5.webdriver.io/docs/api/browser/executeAsync.html) makes it look really easy, eg: `done(a + b + c + d)` - no promises no problem!